pkg/apis/batch/validation/validation.go

```diff
@@ -568,7 +568,7 @@ func validateJobStatus(job *batch.Job, fldPath *field.Path, opts JobStatusValida
 			}
 		}
 	}
-	if ptr.Deref(job.Spec.CompletionMode, batch.NonIndexedCompletion) != batch.IndexedCompletion && isJobSuccessCriteriaMet(job) {
+	if !opts.AllowForSuccessCriteriaMetInExtendedScope && ptr.Deref(job.Spec.CompletionMode, batch.NonIndexedCompletion) != batch.IndexedCompletion && isJobSuccessCriteriaMet(job) {
```

Comment:
I expected to see a validation change that corresponds to the verbiage here: https://github.com/kubernetes/enhancements/blob/master/keps/sig-apps/3939-allow-replacement-when-fully-terminated/README.md?plain=1#L419-L421

that indicates that jobs cannot be `Failed` or `Complete` unless `.status.terminating` is zero.

Reply:
This is implemented under https://github.com/kubernetes/kubernetes/pull/125849 . I initially separated the PRs caused they touched different code. However, given how closely they are connected conceptually I think it makes sense to combine. 

I tentatively put them together by cherry-pick of the commit https://github.com/kubernetes/kubernetes/pull/125510/commits/70c49652704b3dba47b77888f53d34f20cbfcf0b.

PTAL @alculquicondor @soltysh @deads2k if you agree. Otherwise I will return to the idea of 2 PRs.

Reply:
sgtm